mne/io/utils.py

```diff
@@ -249,3 +249,17 @@ def _synthesize_stim_channel(events, n_samples):
     for onset, duration, trigger in events:
         stim_channel[onset:onset + duration] = trigger
     return stim_channel
+
+def _construct_bids_filename(base, ext, part_idx):
+    """Helper function to construct a BIDS compatible filename."""
+    # insert index in filename
+    deconstructed_base = base.split('_')
+    bids_supported = ['meg', 'eeg', 'ieeg']
```

Comment:
do you automatically add the `_meg` suffix when writing using "bids" format?

Reply:
the modality comes from mne-bids. below, the modality is appended back to the filename.

Reply:
yes, but then the filename would not be "BIDS compatible" (unless you use mne-bids) as the comment above suggests? not sure what the docstring says but might be worth fixing ...

Reply:
> yes, but then the filename would not be "BIDS compatible" (unless you use mne-bids) as the comment above suggests? not sure what the docstring says but might be worth fixing ...

@jasmainak, this has been discussed here in https://github.com/mne-tools/mne-python/issues/5519#issuecomment-425199768. there is no way to construct a bids filename in mne-python without porting over the code. this here is to get the split naming to correspond to either neuromag or bids style. it is not to create a proper bids filename